lib/spacesuit/cors_middleware.ex

```diff
@@ -0,0 +1,166 @@
+defmodule Spacesuit.CorsMiddleware do
+  require Logger
+  use Elixometer
+
+  @http_server            Application.get_env(:spacesuit, :http_server)
+  @cors_config            Application.get_env(:spacesuit, :cors_config)
+  @supported_http_methods [:GET, :POST, :PUT, :PATCH, :DELETE, :HEAD, :OPTIONS]
+
+  @timed(key: "timed.corsMiddleware-handle", units: :millisecond)
+  def execute(req, env) do
+    case {req[:headers]["origin"], req[:method]} do
```

Comment:
I'd restructure this whole section. Check out http://relistan.com/elixir-thoughts-on-the-with-statement/